cmd/tools/compare/tools_compare_blocks.go

```diff
@@ -73,9 +73,7 @@ func NewToolsCompareBlocksCmd[B firecore.Block](chain *firecore.Chain[B]) *cobra
 
 	flags := cmd.PersistentFlags()
 	flags.Bool("diff", false, "When activated, difference is displayed for each block with a difference")
-	flags.String("bytes-encoding", "hex", "Encoding for bytes fields, either 'hex' or 'base58'")
```

Comment:
Here you replaced with the global `bytes-encoding` that supports more values (like base64), but the implementation in tools_compare_blocks.go does NOT support anything else than `hex` or `base58`, watch out there needs to be a few changes later in this file, in `func Compare(...)`